File: fixtures/scale/ruby/large/lib/check_blockrotate.rb
require_relative 'blockrotate_encrypt'
require_relative 'blockrotate_decrypt'
require_relative 'corpus_rotate'

def check_blockrotate
  corpus_rotate.each do |text|
    sealed = blockrotate_encrypt(text, 16)
    opened = blockrotate_decrypt(sealed, 16)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_blockrotate.py
from cipherlab.ciphers.blockrotate import blockrotate_encrypt, blockrotate_decrypt
from cipherlab.specs.blockrotate_spec import blockrotate_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_blockrotate():
    spec = blockrotate_spec()
    for text in corpus_rotate():
        sealed = blockrotate_encrypt(text, spec["key"])
        opened = blockrotate_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"
